build.roc

```diff
@@ -0,0 +1,153 @@
+app [main] {
+    cli: platform "platform/main.roc",
+}
+
+import cli.Task exposing [Task]
+import cli.Cmd
+import cli.Stdout
+import cli.Env
+import cli.Arg
+import cli.Arg.Opt
+import cli.Arg.Cli
+
+## Builds the basic-cli [platform](https://www.roc-lang.org/platforms).
+##
+## run with: roc ./build.roc --release
+##
+## Check basic-cli-build-steps.png for a diagram that shows what the code does.
+##
+main : Task {} _
+main =
+
+    cliParser =
+        { Arg.Cli.combine <-
+            releaseMode: Arg.Opt.flag { short: "r", long: "release", help: "Release build. Passes `--release` to `cargo build`." },
+            maybeRoc: Arg.Opt.maybeStr { short: "p", long: "roc", help: "Path to the roc executable. Can be just `roc` or a full path."},
+        }
+        |> Arg.Cli.finish {
+            name: "basic-cli-builder",
+            version: "",
+            authors: ["Luke Boswell <https://github.com/lukewilliamboswell>"],
+            description: "Generates all files needed by Roc to use this basic-cli platform.",
+        }
+        |> Arg.Cli.assertValid
+
+    when Arg.Cli.parseOrDisplayMessage cliParser (Arg.list! {}) is
+        Ok args -> run args
+        Err errMsg -> Task.err (Exit 1 errMsg)
+
+run : { releaseMode : Bool, maybeRoc : Result Str err} -> Task {} _
+run = \{ releaseMode, maybeRoc } ->
+
+    rocCmd = maybeRoc |> Result.withDefault "roc"
+
+    generateGlue! rocCmd
+
+    # target is MacosArm64, LinuxX64,...
+    target = getNativeTarget!
+
+    stubLibPath = "platform/libapp.$(stubFileExtension target)"
+
+    buildStubAppLib! rocCmd stubLibPath
+
+    cargoBuildHost! releaseMode
+
+    copyHostLib! releaseMode target
+
+    preprocessHost! rocCmd releaseMode stubLibPath
```

Comment:
I wanted to have a clean list of all the steps, so I split this up into functions.

Reply:
Looks good